lib/create-plugin.js

```diff
@@ -279,4 +283,45 @@ function getDeepestSelectionPath(definition) {
   }
 }
 
+function recordResolveSegment(segment, scope) {
+  const duration = segment.getDurationInMillis()
+  const exclusive = segment.getExclusiveDurationInMillis()
+
+  const transaction = segment.transaction
+
+  const attributes = segment.getAttributes()
+  const fieldName = attributes[FIELD_NAME_ATTR]
```

Comment:
I think this will work with an alias for a field.  minor: add a test with an alias?

Reply:
We don't really have anything considering alias' at all.

Perhaps enter a new issue to explore the impacts of aliases and add tests to the various considerations (transaction naming, segment naming, attributes, metrics).

Reply:
Sounds good 👍